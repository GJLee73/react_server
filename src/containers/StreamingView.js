/**
 * 플레이어 별 스트리밍 페이지. 채팅과 플레이어가 존재.
 * @author G1
 * @logs // 18.2.26
 */

import React from 'react';
import StreamingPlayerContainer from './StreamingPlayerContainer'
import ChattingContainer from './ChattingContainer';

const StreamingView = ({match})=>{ 
	return (
	<div>
		<StreamingPlayerContainer streamName={match.params.streamname}/>
		<ChattingContainer room={match.params.streamname}/>
	</div>
)}

export default StreamingView;