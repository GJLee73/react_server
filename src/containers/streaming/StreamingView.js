/**
 * 플레이어 별 스트리밍 페이지. 채팅과 플레이어가 존재.
 * @author G1
 * @logs // 18.2.26
 */

import React, { Component } from 'react';
import StreamingPlayerContainer from './StreamingPlayerContainer'
import ChattingContainer from './ChattingContainer';
import {getTheNumberOfToken } from '../../modules/donation';
import {connect} from 'react-redux'
import Donation from './Donation';
import styled from 'styled-components'


export class StreamingView extends Component {
	constructor(props) {
		super(props);
		this.state = { donationModal:false }
		this.donationToggle= this.donationToggle.bind(this)

	}
	/**
	 * 도네이션을 토글하는 함수.
	 */
	donationToggle(){
		if(!this.state.donationModal){
			console.log('222')
			this.props.getTheNumberOfToken()
		}
		console.log('toggle')
		this.setState({
			donationModal: !this.state.donationModal
		  });
	}
	
	render() { 
		return( 
		<div className='row' style={{height:'calc(100vh - 105px)'}}>
			<StreamingPlayerContainer streamName={this.props.match.params.streamname}/>
			<ChattingContainer room={this.props.match.params.streamname} donationToggle={this.donationToggle}/>
			{/* TODO:도네이션 처리를 위한 함수, 리덕스 생성. */}
			<Donation modal={this.state.donationModal} toggle={this.donationToggle} streamName={this.props.match.params.streamname}/>
		</div> )
	}
}

const mapDispatchToProps=(dispatch)=>{
	return{
		getTheNumberOfToken:()=>dispatch(getTheNumberOfToken())
	}
	
}
export default connect(undefined,mapDispatchToProps)(StreamingView)
