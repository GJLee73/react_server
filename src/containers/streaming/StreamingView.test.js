import React from 'react';
import { shallow} from 'enzyme';
import {StreamingView} from './StreamingView'


/**
 * streamingView의 하위 컴포넌트가 store를 포함하고 있기 때문에 mount사용 불가.
 */
describe('Streaming veiw test',()=>{
var wrapper =  shallow(<StreamingView match={{params:{streamname:'a'}}}/>);
  it('snapshot test',()=>{
    expect(wrapper).toMatchSnapshot()
  })
})
