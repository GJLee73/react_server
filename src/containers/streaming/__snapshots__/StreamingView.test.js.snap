// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Streaming veiw test snapshot test 1`] = `
<div
  className="row"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <StreamingPlayerContainer
    streamName="a"
  />
  <Connect(ChattingContainer)
    donationToggle={[Function]}
    room="a"
  />
  <Connect(Donation)
    modal={false}
    streamName="a"
    toggle={[Function]}
  />
</div>
`;
